### PR TITLE
{Databricks} Migrate databricks with CodeGen v2

### PR DESCRIPTION
**Pull request by ZengTaoxu**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Databricks

File: src/databricks/HISTORY.rst [hunk 1]
```diff
@@ -2,6 +2,11 @@
 
 Release History
 ===============
+0.9.0
++++++
+* az databricks workspace and az databricks workspace vnet-peering: Migrate to atomic commands
```
Comment (necusjz) [inline src/databricks/HISTORY.rst:7]: "Command is atomic or not" is transparent to users, so we don't need to declare it in changelog.

File: src/databricks/HISTORY.rst [hunk 2]
```diff
@@ -2,6 +2,11 @@
 
 Release History
 ===============
+0.9.0
++++++
+* az databricks workspace and az databricks workspace vnet-peering: Migrate commands
```
Comment (necusjz) [inline src/databricks/HISTORY.rst:7]:
```suggestion
* az databricks workspace and az databricks workspace vnet-peering: Migrate commands
* Upgrade API version from 2021-04-01-preview to 2022-04-01-preview
```

**Review by jsntcy — APPROVED** [review]

File: src/databricks/azext_databricks/azext_metadata.json [hunk 3]
```diff
@@ -1,3 +1,3 @@
 {
-    "azext.minCliCoreVersion": "2.3.1"
-}
+    "azext.minCliCoreVersion": "2.42.0"
```
Comment (kairu-ms) [inline src/databricks/azext_databricks/azext_metadata.json:2]:
```suggestion
    "azext.minCliCoreVersion": "2.43.0"

```

File: src/databricks/azext_databricks/custom.py [hunk 4]
```diff
@@ -7,170 +7,32 @@
...
-                       workspace_name=workspace_name,
-                       peering_name=peering_name)
-
-
-def delete_databricks_vnet_peering(client, resource_group_name, workspace_name, peering_name, no_wait=False):
-    return sdk_no_wait(no_wait, client.begin_delete,
-                       resource_group_name=resource_group_name,
-                       workspace_name=workspace_name,
-                       peering_name=peering_name)
+        if not args.managed_resource_group:
```
Comment (kairu-ms) [inline src/databricks/azext_databricks/custom.py:31]:
```suggestion
        if not has_value(args.managed_resource_group):

```

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**